### ISSUE TITLE
Thumbnails not working in Windows 11.

### ISSUE THREAD

**Issue by odyskat** [f3d-app/f3d#1129]
**Describe the bug**
Thumbnails (for step files at least) don't work on Windows 11.
(I did not check other file types.)

**To Reproduce**
 1. Installed f3d through convention .exe file from https://f3d.app/doc/user/INSTALLATION.html
 2. Checked that thumb-nailing option is enabled in installation wizard. 
 3. F3d works and opens step file

**System Information:**
Edition	Windows 11 Business
Version	23H2
Installed on	‎11/‎22/‎2023
OS build	22631.2861
Experience	Windows Feature Experience Pack 1000.22681.1000.0

Processor	Intel(R) Core(TM) i9-14900KF   3.20 GHz

[Display]
DirectX version:	12.0 
GPU processor:		NVIDIA GeForce RTX 4070

**F3D Information**
F3D-2.2.1-Windows-x86_64-raytracing

**Comment by mwestphal** [comment]
Hi @odyskat 

I have no access to a Windows 11 computer at the moment so I cant test this right away.
However I see a lot of posts online about general issues with thumbnails on Windows 11. Can you confirm thumbnaisl are working in general for you ?

**Comment by odyskat** [comment]
Hi @mwestphal and thanks for the quick reply.
I checked for a stl file and 
I think f3d did not provide a thumbnail (I see a picture like the one below even when I uninstalled it).
I suppose windows provides the image and not f3d.
Hope it helps.
![Screenshot 2023-12-27 111322](https://github.com/f3d-app/f3d/assets/114591654/ee229d0e-981a-40e8-8a0e-2253acc361be)

**Comment by mwestphal** [comment]
@Meakk tested with Windows 11 and it seems to work great. can you try reinstalling ?

**Comment by odyskat** [comment]
@mwestphal I did it multiple times, same result.
I will try it at home with another computer for comparison.

**Comment by odyskat** [comment]
I tried in a Windows 10 machine at home with no luck for the thumbnails.
Step files were exported from 2 different cad programs to make sure that the problem is not file related.

Note: I use Greek as system locale, not s long ago a Lenovo tool refused to work if the system locale was not English us.
It was hard to find and probably not connected to the thumbnailing issue.
![image](https://github.com/f3d-app/f3d/assets/114591654/15b63510-e94f-4a8c-9735-b629c31ad135)

**Comment by Meakk** [comment]
Maybe it's related to one drive? Can you try on a local disk?

**Comment by odyskat** [comment]
Ok, I tried many cases :

Number 1: win10 folder not connected to OneDrive --> works!

![WhatsApp Image 2024-01-03 at 09 19 37_df500cbd](https://github.com/f3d-app/f3d/assets/114591654/a352f752-45dc-47b2-a69c-13bcfcac3bc5)

Number 2: Win 11 folder not connected to OneDrive --> works!

![image](https://github.com/f3d-app/f3d/assets/114591654/19b2f71e-b061-4d06-afc0-10c422845746)

Number 3: win11 folder in OneDrive fist time copying of the file works, when closed and reopened it's gone.

![image](https://github.com/f3d-app/f3d/assets/114591654/ca41a208-f9d1-4e4d-9194-37005a604694)
![image](https://github.com/f3d-app/f3d/assets/114591654/508c878c-2972-4421-bcd8-19ba835b3524)
![image](https://github.com/f3d-app/f3d/assets/114591654/b3cdb6af-40e0-4f58-8b5e-054a51c4c36a)

**Comment by odyskat** [comment]
Comment : I noticed that when the file is stored in SharePoint (for anyone who hasn't used this, it's like Microsoft's business version of OneDrive) and also saved locally on the computer, thumbnails seem to work fine.

**Comment by Meakk** [comment]
I think OneDrive might remove thumbnails to save space (they are saved in `thumbs.db` file)
I came across this page: https://answers.microsoft.com/en-us/windows/forum/all/video-thumbnails-not-generating-in-onedrive-folder/bf54402a-2306-444a-a986-4642430f6b95  
They are mentioning an option `Files On-Demand: Save space and download files as you use them` to disable. Can you try that?

**Comment by jeedee23** [comment]
I have the same install, and no thumbnails what so ever are showing for step files, but whenI copy the complete folder to the C drive: this happens: 
![afbeelding](https://github.com/f3d-app/f3d/assets/38980341/b7164274-dfcd-4dbc-852e-a9182d3c7b66)
and then the thumbnails appear. one by one.
So, when you have a onedrive, it downloads the thumbnails from onedrive (just do the 'see online' option, and you will see which files have thumbnails, and which not) same for PDF, and others, if not local but onedrive (even if 'always keep on this device' is checked), no thumbnail which MS supports)
You have to sync differently with onedrive.

**Comment by Meakk** [comment]
Regarding the error in the window, it's a new issue.  
@mwestphal see https://github.com/f3d-app/f3d/blob/2f95ee30cc1d36f5f522558656926d13ed952dd9/winshellext/F3DThumbnailProvider.cxx#L168

**Comment by Meakk** [comment]
@jeedee23 I think the files without thumbnails are probably the bigger files, can you confirm?

**Comment by mwestphal** [comment]
fix incoming, but I think @jeedee23  is using a old nightly.

https://github.com/f3d-app/f3d/pull/1151

**Comment by jeedee23** [comment]
Yes, but the new latest, doesn’t give any thumbnails

 

Van: Mathieu Westphal ***@***.***> 
Verzonden: Tuesday, 9 January 2024 22:58
Aan: f3d-app/f3d ***@***.***>
CC: Johan Degraeve ***@***.***>; Mention ***@***.***>
Onderwerp: Re: [f3d-app/f3d] Thumbnails not working in Windows 11 with OneDrive (Issue #1129)

 

fix incoming, but I think @jeedee23 <https://github.com/jeedee23>  is using a old nightly.

#1151 <https://github.com/f3d-app/f3d/pull/1151>

**Comment by jeedee23** [comment]
Hi Michael, yes, bigger

 

Van: Michael MIGLIORE ***@***.***> 
Verzonden: Tuesday, 9 January 2024 22:58
Aan: f3d-app/f3d ***@***.***>
CC: Johan Degraeve ***@***.***>; Mention ***@***.***>
Onderwerp: Re: [f3d-app/f3d] Thumbnails not working in Windows 11 with OneDrive (Issue #1129)

 

@jeedee23 <https://github.com/jeedee23>  I think the files without thumbnails are probably the bigger files, can you confirm?

**Comment by Meakk** [comment]
@jeedee23 If you look at your installation directory, you will find a `share\f3d\configs\thumbnail.d\10_global.json` config file.  
This config file sets the max size limit for generating thumbnails. By default, it's 100MB: `"max-size":100`.  
You can increase this value directly here, but it's recommended to create your own "user" config file to avoid an overwrite during a f3d update, please refer to https://f3d.app/doc/user/CONFIGURATION_FILE.html

**Comment by Meakk** [comment]
> Yes, but the new latest, doesn’t give any thumbnails

Is it expected @mwestphal?

**Comment by mwestphal** [comment]
||   Yes, but the new latest, doesn’t give any thumbnails

| Is it expected @mwestphal?

Definitely not, but we will need actual version numbers from @jeedee23 to move forward. Testing the last nightly on windows would also be a start.

**Comment by odyskat** [comment]
> I think OneDrive might remove thumbnails to save space (they are saved in `thumbs.db` file) I came across this page: https://answers.microsoft.com/en-us/windows/forum/all/video-thumbnails-not-generating-in-onedrive-folder/bf54402a-2306-444a-a986-4642430f6b95 They are mentioning an option `Files On-Demand: Save space and download files as you use them` to disable. Can you try that?

I don't have enough space in my work comp.
But I tried with one folder, same issue.
I updated to Windows 11, version 23H2. Now, new files downloaded from SharePoint locally do not want to display thumbnails, but files I had downloaded still have thumbnails.

**Comment by jeedee23** [comment]
> || Yes, but the new latest, doesn’t give any thumbnails
> 
> | Is it expected @mwestphal?
> 
> Definitely not, but we will need actual version numbers from @jeedee23 to move forward. Testing the last nightly on windows would also be a start.

F3D-2.2.1-95-g2f95ee3-Windows-x86_64 is the version I use. The only one giving thumbnails for step files, but not for all of them (even if size < 100MB) **and it shows the pop-up**

**Comment by jeedee23** [comment]
> I think OneDrive might remove thumbnails to save space (they are saved in `thumbs.db` file) I came across this page: https://answers.microsoft.com/en-us/windows/forum/all/video-thumbnails-not-generating-in-onedrive-folder/bf54402a-2306-444a-a986-4642430f6b95 They are mentioning an option `Files On-Demand: Save space and download files as you use them` to disable. Can you try that?

This is an old thread and this option is not present anymore.

**Comment by mwestphal** [comment]
I just tested the last nightly and thumbnails are working fine (outside of onedrive which I do not use).

@Meakk did you actually try with onedrive ?

**Comment by jeedee23** [comment]
Yes it is a onedrive issue, other apps have the same problem. The files are not stored the usual way, I think there is need for a MS api to be able to do this

**Comment by Meakk** [comment]
> @Meakk did you actually try with onedrive ?

No I don't have a onedrive account

**Comment by mwestphal** [comment]
>  You have to sync differently with onedrive.

Ok, looks like we do need to do something else to support onedrive file. Which we will probably not do for now I'm afraid.

Related to: #797

**Comment by jeedee23** [comment]
> > You have to sync differently with onedrive.
> 
> Ok, looks like we do need to do something else to support onedrive file. Which we will probably not do for now I'm afraid.
> 
> Related to: #797

Yes, that is what I thought, because it is almost impossible.

**Comment by mwestphal** [comment]
labelling as wont fix.

**Comment by AlanEdd** [comment]
FYI to assist, 
I had trouble with icaros icon generation locally with OneDrive where OneDrive would do the same thing listed here, ie. delete the existing thumbnail. I checked everything and a year later on a single OneDrive update, everything just worked. It was working previously, stopped working for 6 months of OneDrive updates and then was rectified. Nothing changed with Icaros. I have F3d installed here, so will shout if something changes.

**Comment by Meakk** [comment]
I investigated a bit. So it seems like files visible in the file explorer stored in OneDrive are called ["placeholders"](https://learn.microsoft.com/en-us/windows/win32/w8cookbook/placeholder-files). It contains only metadata and a thumbnail **generated on the server**.  
Unfortunately, I don't think there's anything we can do, it's Microsoft responsibility. It seems like there's an API to force "hydration" (download) of the files, but I don't think it's a good idea.  
@mwestphal should we close this?

**Comment by mwestphal** [comment]
> but I don't think it's a good idea.

Agreed.

**Comment by AlanEdd** [comment]
Ok but just a point, 
Icaros generates thumbnails fine on my machine and they seem to be preserved. That's for all the files Icaros handles on OneDrive.

**Comment by mwestphal** [comment]
> Ok but just a point, Icaros generates thumbnails fine on my machine and they seem to be preserved. That's for all the files Icaros handles on OneDrive.

This project ?

https://github.com/Xanashi/Icaros

Do you know how they handle it ?

**Comment by AlanEdd** [comment]
Yes that's it, seems the thumbnails stick, I don't have the knowledge how they do it. It failed for about a year but OneDrive changed something and they came back in an update
